src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/ExpressionJsonConverter.cs

```diff
@@ -0,0 +1,43 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class ExpressionJsonConverter : JsonConverter
```

Comment:
Is the reason for the converter for backwards compatibility with existing templates?

Reply:
If so it would be good to add comment to this class to explain its use.

Reply:
Its a custom converter which allows use to build `Expression` objects that are represented as either String or JObjects. For example, it supports the below Json snippets:

```
 {
  "typeMatchExpression" : "@.heartRate"
}
```

and 

```
 {
  "typeMatchExpression" : {
       "value" : "@.heartRate",
       "language" : "JsonPath"
  }
}
```